doc/source/serve/production.md

```diff
@@ -583,3 +565,217 @@ This section summarizes the best practices when deploying to production:
+
+```console
+  - name: MangoStand
+    numReplicas: 1
+    userConfig: |
+      price: 4
+```
+
+Now to update the application we apply the modified manifest:
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  serveDeploymentStatuses:
+  - healthLastUpdateTime: "2022-07-18T21:51:37Z"
+    lastUpdateTime: "2022-07-18T21:51:41Z"
+    name: MangoStand
+    status: UPDATING
+...
+```
+
+If we query the application, we can see that we now get a different result reflecting the updated price:
+
+```console
+$ curl -X POST -H 'Content-Type: application/json' localhost:8000 -d '["MANGO", 2]'
+8
+```
+
+#### Updating the RayCluster config
+
+The process of updating the RayCluster config is the same as updating the Serve config.
+For example, we can update the number of worker nodes to 2 in the manifest:
+
+```console
+workerGroupSpecs:
+  # the pod replicas in this group typed worker
+  - replicas: 2
+```
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl describe
+...
+  pendingServiceStatus:
+    appStatus: {}
+    dashboardStatus:
+      healthLastUpdateTime: "2022-07-18T21:54:53Z"
+      lastUpdateTime: "2022-07-18T21:54:54Z"
+    rayClusterName: rayservice-sample-raycluster-bshfr
+    rayClusterStatus: {}
+...
+```
+
+In the status, you can see that the `RayService` is preparing a pending cluster.
+After the pending cluster is healthy, it is switched to the active cluster and the previous cluster is terminated.
+
+## Failure Recovery
+
+Ray Serve is resilient to any component failures within the Ray cluster out of the box.
+You can read more about how worker node and process failures are handled in: {ref}`serve-ft-detail`.
+
+### Head node failures
+
+By default the Ray head node is a single point of failure: if it crashes, the entire cluster crashes and needs to be restarted.
+When running on Kubernetes, the `RayService` controller health checks the cluster and restarts it if this occurs, but this still results in some downtime.
+In Ray 2.0, `KubeRay` has added experimental support for [GCS fault tolerance](https://ray-project.github.io/kuberay/guidance/gcs-ha/#ray-gcs-ha-experimental), preventing the Ray cluster from crashing if the head node goes down.
+While the head node is recovering, Serve applications can still handle traffic but cannot be updated or recover from other failures (e.g., actors or worker nodes crashing).
+Once the GCS is recovered, the cluster will return to normal behavior.
+
+### Health checking
+
+By default, each Serve deployment replica is periodically health checked and restarted on failure.
+
+You can customize this behavior to perform an application-level health check or to adjust the frequency/timeout.
+To define a custom healthcheck, define a `check_health` method on your deployment class.
+This method should take no arguments and return no result, and it should raise an exception if the replica should be considered unhealthy.
```

Comment:
Can users view this error after failure? If so, can we explain how in this section?

Users might propagate info in the error message to debug why the health check failed, so it would be nice if they could see the message.

Reply:
I don't think the exception is propagated beyond the logs right now. this would be a good future improvement

Reply:
i'll make a note